plugin/resolve/resolve.go

```diff
@@ -0,0 +1,123 @@
+package resolve
+
+import (
+	"context"
+	"strings"
+
+	"github.com/coredns/coredns/plugin"
+	"github.com/coredns/coredns/plugin/pkg/nonwriter"
+	"github.com/coredns/coredns/request"
+
+	"github.com/miekg/dns"
+)
+
+// Resolve performs CNAME target resolution on all CNAMEs in the response
+type Resolve struct {
+	Next    plugin.Handler
+	Zones   []string
+	DoCNAME bool
+	DoSRV   bool
+}
+
+// Name implements the Handler interface.
+func (c Resolve) Name() string { return name() }
+func name() string             { return "resolve" }
```

Comment:
why this indirection?

Reply:
Allows us to define the literal string "resolve" once.  It's a pattern followed in a few other plugins. It could also be a constant... or just have the literal name of the plugin pasted in all over the place like we do in most plugins.  IMO many valid ways to do the same thing.